packages/components/src/components/data-grid/data-grid.tsx

```diff
@@ -416,15 +420,21 @@ export class DataGrid {
     this.activeSortingIndex = columnIndex;
 
     const newSortDirection =
-      sortDirection === 'none'
+      currentSortDirection === 'none'
         ? 'ascending'
-        : sortDirection === 'ascending'
+        : currentSortDirection === 'ascending'
         ? 'descending'
         : 'none';
     this.fields[columnIndex].sortDirection = newSortDirection;
     this.sortTable(newSortDirection, type, columnIndex);
   }
 
+  presortTable(sortDirection, columnIndex, type): void {
```

Comment:
Why not send the `column`  itself?  Since you are finding an index once in the presort function and finding the field here again. We can skip that part

Reply:
this logic is used initially in template and other sorting methods, changing causes changes across the whole file, but I can)